src/components/button/config/generate.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright IBM Corp. 2018, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Template configuration generate functions for Button component
+ * @module button/generate
+ */
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+
+// TODO JSDoc
+export function generate(options) {
```

Comment:
I think I'm confused by the `generate` function being in the `spec`. My thoughts were that an individual repo would take care of this level of integration. i.e. the simpler api `{variant: 'primary'}` would be what is shared by c-c-X, and each repo would be in charge of converting that to the proper template configuration object for their implementation. 

open to new thinking though

Reply:
I think it depends on the two options stated [here](https://github.com/carbon-design-system/carbon-spec/issues/6#issuecomment-458723207)?

This function here isn't meant to be a full template configuration object - although maybe it's close enough to act as that for Handlebars? The goal of the configuration object would be to give each component API just enough for it to render an accessible component.